src/Microsoft.Kusto.ServiceLayer/DataSource/Metadata/FolderMetadata.cs

```diff
@@ -5,6 +5,6 @@
     /// </summary>
     public class FolderMetadata : DataSourceObjectMetadata
     {
-        public DataSourceObjectMetadata ParentMetadata { get; set; }
```

Comment:
Renamed since this contains the database metadata and not the immediate parent.

Reply:
What about the nested folder case? I think we should leave this generic for adding other datastores. It might be database for Kusto. But it could be completely something else for other stores. Better to leave it as "ParentMetadata" and let the store implementation decide what the parent refers to. The Parent here is meant to be the parent in the hierarchy for the object explorer.